### DISCUSSION TITLE
Validation child component

### DISCUSSION THREAD

**Discussion by nachoalcazar** [quasarframework/quasar#11888] [General - Components / Directives / etc] [answered]
Hello, I've a child component:

All works OK excepts external validation. (autovalidation works OK)

DatePickerCustom.js
```
<template>
  <q-input
    :model-value="modelValue"
    filled
    :mask="maskFechaCorta"
    :rules="[validarFecha]"
    class="no-margin no-padding"
    @update:modelValue="event=>$emit('update:modelValue', event)"
  >
...
</q-input>
</template>

...

export default {
  props: {
    modelValue: {
      type: [String],
      default: ''
    },
...
  },
  emits: ['update:modelValue'],
...

```

And when I try to validate him from de parent USING **ref**: 

view.js

```
...
                <date-picker-custom
                  ref="reffecha"
                  v-model="fecha"
                  label="Fecha de compra"
                />
```

```
setup() {
...
    const reffecha = ref(null)
...
 const method = () => {
if (reffecha.value.validate()) {
...
}
}
```

TypeError: reffecha.value.validate is not a function

console.log(reffecha.value):

![image](https://user-images.githubusercontent.com/94051561/148036126-34472435-c4c9-4ad3-b897-65d4487d06ab.png)

When I haven't child component and I has de q-input/q-date into the parent class, works OK
Any suggestion?

**Comment by pdanpdan** [comment, marked as answer]
You need to expose the `validate` and `resetValidation` methods of the QInput on your wrapper.
https://quasar.dev/vue-components/form#child-communication

**Reply by nachoalcazar** [reply]
Thanks. All works OK.

I thought i could call the method validate of q-input, not directly my validate method (is the same anyway)

**Reply by xnagaev** [reply]
@nachoalcazar @pdanpdan could You help me to figure it out how to force validate child component from parent?

**Reply by nachoalcazar** [reply]
First post:

```
<template>
     <date-picker-custom
                  ref="reffecha"
                  v-model="fecha"
                  label="Fecha de compra"
                />

</template>


<script setup>

...
    const reffecha = ref(null)
...
 const method = () => {
if (reffecha.value.validate()) {
...
}
```

**Comment by pdanpdan** [comment]
you have to forward them from QInput to the wrapper
